kairos-server/src/lib.rs

```diff
@@ -26,8 +28,37 @@ pub fn app_router(state: Arc<state::BatchStateManager>) -> Router {
         .with_state(state)
 }
 
+pub async fn run_l1_sync(config: ServerConfig, batch_service: Arc<BatchStateManager>) {
+    // Make sure real contract hash was provided.
+    if config.casper_contract_hash
+        == "0000000000000000000000000000000000000000000000000000000000000000"
+    {
+        tracing::warn!(
+            "Casper contract hash not configured, L1 synchronization will NOT be enabled."
+        );
+        return;
+    }
```

Comment:
I am expecting that contract hash is always required - same as the RPC url. Is there any reason to run Kairos server without observing L1 state? I think no, so I introduced default value `00...00` with warning if unchanged.

Another reason to avoid `Option` for config field, is that it requires creating newtype with  `FromStr` implementation. IMO code gets unnecessarily complicated, you can see changes preview here - https://github.com/cspr-rad/kairos/commit/74adecdd9959aa0dc5364af3ef0abe8ed8a7c1ff. I would prefer to avoid it, but let me know if you like it better.